DBZ-5762 Update Confluent Avro usage documentation (#4024)
debezium/debezium: documentation/modules/ROOT/pages/configuration/avro.adoc

For information about setting up this registry, see the documentation for link:{LinkServiceRegistryInstall}[{NameServiceRegistryInstall}].
endif::product[]

// Type: concept
// Title: {registry}
[id="apicurio-registry"]
== {registry-name-full}

// Type: concept
// Title: About the {registry}
[id="about-the-registry"]

> === About the {registry-name-full}

Reviewer: I don't understand the logic behind adding the redundant title heading here, as it doesn't really add anything.  I suggest eliminating lines 52-56 in favor of lines 57-60, and restoring the second level heading level on L60.

Author: There are two independent schema registry - Apicurio and Confluent. Currently sub-chapters regarding Apicurio and Confluent are on the same level without any separation, which can be harder to read and understand, making the feel that they are the same thing, which is not true. This adds higher level chapter `Apicurio API and Schema Registry` and later on this PR also adds `Confluent Schema Registry` chapter which separates these two technologies into dedicated sections, logically separating these two independent technologies. So instead of current

* About the Apicurio API and Schema Registry
* Deployment overview
* Deploying with Debezium containers
* Confluent Schema Registry
* Naming
* Getting More Information

the structure now is as follows:

* Apicurio API and Schema Registry
  * About the Apicurio API and Schema Registry
  * Apicurio Registry deployment overview
  * Deploying Apicurio Registry with Debezium containers
* Confluent Schema Registry
  * Confluent Schema Registry deployment overview
  * Deploying Confluent Schema Registry with Debezium containers
* Naming
* Getting More Information

endif::product[]

ifdef::community[]

> . Install the Avro converter from link:https://repo1.maven.org/maven2/io/apicurio/apicurio-registry-distro-connect-converter/{apicurio-version}/apicurio-registry-distro-connect-converter-{apicurio-version}.tar.gz[the installation package] into a plug-in directory. This is not needed when using the link:https://quay.io/repository/debezium/connect[Debezium Connect container image], see details in <<deploying-with-debezium-containers>>.

Reviewer: ```suggestion
. Install the Avro converter from the link:https://repo1.maven.org/maven2/io/apicurio/apicurio-registry-distro-connect-converter/{apicurio-version}/apicurio-registry-distro-connect-converter-{apicurio-version}.tar.gz[installation package] into a plug-in directory. If you are using the link:https://quay.io/repository/debezium/connect[Debezium Connect container image], it's not necessary to install the package. For more information, see  <<deploying-with-debezium-containers>>.
```

== Confluent Schema Registry

There is an alternative https://github.com/confluentinc/schema-registry[schema registry] implementation provided by Confluent.

[id="overview-of-deploying-confluent-schema-registry"]
=== Confluent Schema Registry deployment overview

> To install standalone Confluent Schema Registry, please see Confluent's https://docs.confluent.io/platform/current/installation/installing_cp/overview.html[official deployment guide].

Reviewer: ```suggestion
For information about installing the standalone Confluent Schema Registry, see the  https://docs.confluent.io/platform/current/installation/installing_cp/overview.html[Confluent Platform deployment documentation].
```

> [NOTE]
> ====
> There's also possibility to https://docs.confluent.io/platform/current/installation/docker/installation.html[install] it as a container.
> ====

Reviewer: ```suggestion
As an alternative, you can https://docs.confluent.io/platform/current/installation/docker/installation.html[install] the standalone Confluent Schema Registry as a container.
```
I would add this information in a standard paragraph, rather than presenting it in a note.

[id="deploying-confluent-schema-registry-with-debezium-containers"]
=== Deploying Confluent Schema Registry with {prodname} containers

> As of {prodname} 2.0.0, Confluent Schema Registry support is on part of the {prodname} containers.

Reviewer: ```suggestion
Beginning with {prodname} 2.0.0, Confluent Schema Registry support is included in the {prodname} containers.
```
It's not clear what you mean when you state that in 2.0 CSR support "is on part of the Debezium containers." I'm interpreting this to mean that the CSR registry is built in to the Debezium 2.0 containers, yes?

Author: ops, what I wrote don't make any sense. Actually I meant that the support of Confluent Schema Registry was dropped. i adjusted your suggestion.

> To make it working, you need to install into Connect plugin directory appropriate jar files which contain Confluent Avro converters.
> Namely, you would need following jar file:
> 
> * `kafka-connect-avro-converter`
> * `kafka-connect-avro-data`
> * `kafka-avro-serializer`
> * `kafka-schema-serializer`
> * `kafka-schema-registry-client`
> * `common-config`
> * `common-utils`

Reviewer: ```suggestion
To enable the Confluent Schema Registry for a {prodname} container, install the following Confluent Avro converter JAR files into the Connect plugin directory:
```

* `kafka-connect-avro-converter`
* `kafka-connect-avro-data`
* `kafka-avro-serializer`
* `kafka-schema-serializer`
* `kafka-schema-registry-client`
* `common-config`
* `common-utils`
```

> [NOTE]
> ====
> You can download the required jar files https://packages.confluent.io/maven/[Confluent's Maven repository].
> ====

Reviewer: ```suggestion
You can download the preceding files from the https://packages.confluent.io/maven/[Confluent Maven repository].
```
Similar to my earlier comment, you can drop the note formatting here and just add this information in-line as a standard paragraph.

== Confluent Schema Registry

There is an alternative https://github.com/confluentinc/schema-registry[schema registry] implementation provided by Confluent.

// Type: concept

> // Title: Overview of deploying a Confluent Schema Registry

Reviewer: These annotation comments are intended for use with content that is exposed downstream and should not be used for community-only content. Unless the Confluent integration is something that we intend to expose in the product, these comments should be removed.

Author: done

[id="overview-of-deploying-confluent-schema-registry"]
=== Confluent Schema Registry deployment overview

To install standalone Confluent Schema Registry, please see Confluent's https://docs.confluent.io/platform/current/installation/installing_cp/overview.html[official deployment guide].

[NOTE]
====
There's also possibility to https://docs.confluent.io/platform/current/installation/docker/installation.html[install] it as a container.
====

// Type: procedure
// Title: Deploying connectors that use Confluent Schema Registry in {prodname} containers

> // ModuleID: deploying-connectors-that-use-avro-in-debezium-containers

Reviewer: Same comment as above.

Author: done
